### PR TITLE
[RLlib] SAC, RNNSAC, and CQL TrainerConfig objects

### PR DESCRIPTION
**Pull request by smorad**
## Why are these changes needed?
This makes SAC, RNNSAC, and CQL use TrainerConfig objects instead of untyped config dicts
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by smorad** [comment]
Failing tests seem to be unrelated, except for https://buildkite.com/ray-project/ray-builders-pr/builds/32986#620a984a-b36e-4e28-b7fb-93b35bca2975, which is related to `RNNSAC`. I believe this is a known issue and should be fixed once https://github.com/ray-project/ray/pull/24923 lands.

**Comment by sven1977** [comment]
Thanks for the PR Steven (@smorad ) :)
Yeah, RNNSAC tests are under observation from @ArturNiederfahrenhorst . He has another PR that fixes the problem. No worries.

**Comment by ArturNiederfahrenhorst** [comment]
> Thanks for the PR Steven (@smorad ) :) Yeah, RNNSAC tests are under observation from @ArturNiederfahrenhorst . He has another PR that fixes the problem. No worries.

Yep, the issue is fixed but fixing it exposed an issue with R2D2 which I'm now fixing in the same PR. Sorry about the lag guys.

File: rllib/agents/trainer_config.py [hunk 1]
```diff
@@ -849,7 +849,7 @@ def evaluation(
             self.evaluation_num_workers = evaluation_num_workers
         if custom_evaluation_function is not None:
             self.custom_evaluation_function = custom_evaluation_function
-        if self.always_attach_evaluation_results:
+        if always_attach_evaluation_results:
```
Comment (sven1977) [inline rllib/agents/trainer_config.py:852]: Wow, nice catch!

**Review by sven1977 — APPROVED** [review]
Looks solid! Thanks Steven.

I made only some very small changes to backward-support policy_model and Q_model and fixed some LINTing stuff.

Let's wait for tests to pass ...

**Pull request merged**